docs/tsg.md

```diff
@@ -112,14 +112,16 @@ ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
 <a name="random_404_returned_for_client_requests"></a>
 ## 404 returned for client requests
 
-For a SignalR persistent connection, it first `/negotiate` to Azure SignalR service and then establish the real connection to Azure SignalR service. Our load balancer must ensure that the `/negotiate` request and the following connect request goes to the similar instance of the Service otherwise 404 occurs. Our load balancer now relies on the *signature* part of the generated `access_token` to keep the session sticky.
+For a SignalR persistent connection, it first `/negotiate` to Azure SignalR service and then establishes the real connection to Azure SignalR service. Our load balancer must ensure that the `/negotiate` request and the following connect request goes to the same instance of the Service otherwise 404 occurs. Our load balancer relies on the *asrs_request_id* query string (added in SDK version 1.0.11) or the *signature* part of the generated `access_token`(if `asrs_request_id` query string does not exist) to keep the session sticky.
 
 ### Troubleshooting Guide
-1. Following [How to view outgoing requests](#view_request) to get the request from client the the service.
-1. Check the url of the request when 404 occurs. If the url is targeting to your web app, and similar to `{your_web_app}/hubs/{hubName}`, check if the client `SkipNegotiation` is `true`. When using Azure SignalR, client receives redirect url when it first negotiates with the app server. Client should **NOT** skip negotiation when using Azure SignalR.
-1. Check if there are multiple `access_token` inside the outgoing request. Our load balancer is not able to handle duplicate `access_token` correctly, as described in [#346](https://github.com/Azure/azure-signalr/issues/346).
-1. Another 404 can happen when the connect request is handled more than **5** seconds after `/negotiate` is called. Check the timestamp of client request, and open an issue to us if the request to the service has very slow response.
-1. If you find the `/negotiate` request and the following connect request carry different access token through the above steps, the most possible reason is using HttpConnectionOptions.AccessTokenProvider in a **WRONG** way:
+There are two kind of 404 errors with different symptoms.
+1. One symptom is that the 404 errors appear from time to time randomly. For this kind of 404, please check:
```

Comment:
The first kind is not random?

Reply:
The first kind is random because, without the `asrs_request_id` or `signature`, there is still chance that the load balancer routes the request to the correct instance.

Reply:
then what's the difference with the second one from user's perspective..

Reply:
The first one is from time to time random requests failure, the second one is for a short period massive 404..... other periods no 404.... yeah, indeed difficult to tell from user's perspective

Reply:
should be `consistently`